.github/workflows/check_security_vulnerability.yml

```diff
@@ -1,27 +1,37 @@
-# This workflow uses actions that are not certified by GitHub.
-# They are provided by a third-party and are governed by
-# separate terms of service, privacy policy, and support
-# documentation.
-
-name: Check Security Vulnerability
+name: Code Quality and Security
 
 on:
   pull_request:
   push:
     branches:
       - main
+  schedule:
+    - cron: '0 0 * * 0' # Run every Sunday at 00:00 (midnight)
```

Comment:
If we run this on every PR and push to main, do we still need to run it regularly?